internal/expr/scan.go

```diff
@@ -117,29 +117,41 @@ func (re *ResultExpr) Decode(args ...any) (err error) {
 
 		v := reflect.ValueOf(arg)
 		if v.Kind() != reflect.Pointer {
-			return fmt.Errorf("none pointer paramter")
+			return fmt.Errorf("none pointer parameter")
 		}
 		v = reflect.Indirect(v)
 
 		re.decodeValue(v)
-
 	}
 
 	return nil
 }
 
 // decodeValue sets the fields in the reflected struct "v" which have tags
-// corrosponding to columns in current row of the query results.
+// corresponding to columns in current row of the query results.
 func (re *ResultExpr) decodeValue(v reflect.Value) error {
 	typeFound := false
 	for i, outDest := range re.outputs {
+		if outDest.typ.Kind() == reflect.Map {
+			if err := CheckValidMapType(outDest.typ); err != nil {
+				return err
+			}
+			if err := CheckValidMapType(v.Type()); err != nil {
```

Comment:
If there ever becomes a way to talk about the `M` type defined in `sqlair.go` here then we could replace this with `outDest.typ == v.Type()`, but for now this is good.

Reply:
With one coming from the `expr` package and the other from `package_test` or `sqlair`, the only way I could think of was to make sure that both had name `M` with keys of type `string`, and calling this method for them individually did the trick. With a proper way to figure out what to do about circular imports, this would be replaced!